src/app/shared/utils/short-hash.pipe.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * This pipe takes a hash (or any string) and shortens it to the first seven characters plus an ellipsis at the end.
+ * @copyright The GHGA Authors
+ * @license Apache-2.0
+ */
+
+import { Pipe, PipeTransform } from '@angular/core';
+
+/**
+ * This pipe is used to shorten a hash or other string to its first seven characters plus an ellipsis at the end.
+ */
+@Pipe({
+  name: 'shortHash',
+})
+export class ShortHash implements PipeTransform {
+  /**
+   * The transform method executes the business logic of the Pipe
+   * @param hash The hash to shorten
+   * @returns The shortened string
+   */
+  transform(hash: string): string {
+    if (hash.length < 7) return hash;
```

Comment:
If the has hash hash has 7 chars, 1234567, then it will become 123456... which is even longer...

I would expect that the output of such a pipe is always shorter than the input.

Reply:
Also, would be good to make the number 7 configurable as a size parameter with default value 7 (you can pass additional params to  pipes),

Reply:
GitHub abbreviates commit hashes with 7 chars, maybe we should also use 7 by default (instead of 6).

Reply:
It is 7 characters, not 6! Seems like a classic off-by-one error 😄 
But I will fix it so it does the ellipsis only for strings longer than 10 characters so it never returns longer strings than the original.

Reply:
Yes, you can add a size parameter with a default of 10 that would guarantee to always output max 10 chars (including the dots). Including the dots, you would have 7 digits from the hash (just like in GitHub).

Aletrnatively, and maybe better, use a single unicode ellipsis char instead of the three dots. Then the default size would be 8.